R/irods-path.R

```diff
@@ -85,7 +85,7 @@ lpath_exists <- function(lpath, write = FALSE) {
   if (!is_connected_irods()) stop("Not connected to iRODS.", call. = FALSE)
 
   # reference paths
-  all_lpaths <- ils(make_irods_base_path(), recurse = 1) |>
+  all_lpaths <- ils(make_irods_base_path(), recurse = 1, limit=NULL) |>
```

Comment:
Why does `ils` take a limit? Does `ils` use the list operation of the /collections endpoint or is it running GenQuery to fetch the results?

The /collections endpoint of the HTTP API will return all entries in a collection.

Reply:
I don't know why it takes a limit, to be honest. Indeed it uses the /collections endpoint, therefore the limit is being implemented post-hoc. Personally, I would remove it altogether.
(At most, it could be an argument for the printing method)

Reply:
Hmm. Perhaps that code used GenQuery in the past to retrieve the listing?

I can see that being the case if more than the list of logical paths is required. The /collections endpoint only returns the paths. No additional information about the paths is included (e.g. permissions, metadata, etc).